src/test/java/org/prebid/server/bidder/gothamads/GothamAdsBidderTest.java

```diff
@@ -0,0 +1,246 @@
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.gothamads.GothamAdsImpExt;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static java.util.function.Function.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.prebid.server.bidder.model.BidderError.Type;
+import static org.prebid.server.bidder.model.BidderError.badServerResponse;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.banner;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.video;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.xNative;
+import static org.prebid.server.util.HttpUtil.ACCEPT_HEADER;
+import static org.prebid.server.util.HttpUtil.APPLICATION_JSON_CONTENT_TYPE;
+import static org.prebid.server.util.HttpUtil.CONTENT_TYPE_HEADER;
+import static org.prebid.server.util.HttpUtil.USER_AGENT_HEADER;
+import static org.prebid.server.util.HttpUtil.X_FORWARDED_FOR_HEADER;
+import static org.prebid.server.util.HttpUtil.X_OPENRTB_VERSION_HEADER;
+import static org.prebid.server.util.HttpUtil.headers;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+
+public class GothamAdsBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://test-url.com/?pass={{AccountId}}";
+    private static final String OPEN_RTB_VERSION = "2.5";
+    private static final String DEVICE_UA = "ua";
+    private static final String IPV_6_HEADER_VALUE = "ipv6";
+    private static final String IP_HEADER_VALUE = "ip";
+    private static final String IMP_ID = "imp_id";
+
+    private GothamAdsBidder sut;
+
+    @BeforeEach
+    public void setUp() {
+        sut = new GothamAdsBidder(ENDPOINT_URL, jacksonMapper);
+    }
+
+    @Test
+    public void shouldFailOnBidderCreation() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new GothamAdsBidder("invalid_url", jacksonMapper));
+    }
+
+    @Test
+    public void shouldReturnErrorWhileMakingHttpRequestsWhenRequestHasInvalidImpression() {
+        // given
+        final ObjectNode invalidExt = mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()));
+        final BidRequest bidRequest = givenBidRequest(impBuilder -> impBuilder.ext(invalidExt));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = sut.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1).first()
+                .satisfies(error -> assertThat(error.getType()).isEqualTo(Type.bad_input));
+    }
+
+    @Test
+    public void shouldMakeCorrectRequest() {
+        // given
+        final String secondImpId = "imp_id2";
+        final BidRequest bidRequest = givenBidRequest(identity(), i -> i.id(secondImpId).ext(givenExt()));
```

Comment:
What is `i`? Please, use meaningful variable names, for example `impBuilder`. Also, please, check for similar occurences.